src/rgw/driver/rados/rgw_rados.cc

```diff
@@ -5130,6 +5130,33 @@ int RGWRados::delete_bucket(RGWBucketInfo& bucket_info, RGWObjVersionTracker& ob
     (void) CLSRGWIssueBucketIndexClean(index_pool.ioctx(),
 				       bucket_objs,
 				       cct->_conf->rgw_bucket_index_max_aio)();
+  } else {
+    bucket_info.flags |= BUCKET_DELETED;
+    static constexpr auto max_retries = 10;
+    int retries = 0;
+    do {
```

Comment:
`bucket_info.flags |= BUCKET_DELETED;` needs to be inside the `do {` so it gets added back after `get_bucket_instance_info()`

Reply:
p.s. it might be worth moving this stuff into a helper function so we don't over-complicate `RGWRados::delete_bucket()`

Reply:
> `bucket_info.flags |= BUCKET_DELETED;` needs to be inside the `do {` so it gets added back after `get_bucket_instance_info()`

huh sorry we are overwriting, yes.